docs/src/examples.md

```diff
@@ -86,6 +86,63 @@ julia> round.(vcov(ols); digits=5)
   0.38889  -0.16667
  -0.16667   0.08333
 ```
+By default, the `lm` method uses the Cholesky factorization which is known as fast but numerically unstable, especially for ill-conditioned design matrices. You can use the `method` keyword argument to apply a more stable QR factorization method.
+
+```jldoctest
+julia> data = DataFrame(X=[1,2,3], Y=[2,4,7]);
+
+julia> ols = lm(@formula(Y ~ X), data; method=:qr)
+LinearModel
+
+Y ~ 1 + X
+
+Coefficients:
+─────────────────────────────────────────────────────────────────────────
+                 Coef.  Std. Error      t  Pr(>|t|)  Lower 95%  Upper 95%
+─────────────────────────────────────────────────────────────────────────
+(Intercept)  -0.666667    0.62361   -1.07    0.4788   -8.59038    7.25704
+X             2.5         0.288675   8.66    0.0732   -1.16797    6.16797
+─────────────────────────────────────────────────────────────────────────
+```
+The following example shows that QR decomposition works better for an ill-conditioned design matrix. The linear model with the Cholesky decomposition method is unable to estimate parameters correctly whereas the linear model with the QR decomposition does.
+Note that, the condition number of the design matrix is quite high (≈ 3.52e7). Since the Cholesky method aggressively detect multicollinearity, if you ever encounter multicollinearity in any GLM model with Cholesky, it is worth trying the same model with QR decomposition. 
+
+```
+julia> X = [-0.4011512997627107 0.6368622664511552;
+            -0.0808472925693535 0.12835204623364604;
+            -0.16931095045225217 0.2687956795496601;
+            -0.4110745650568839 0.6526163576003452;
+            -0.4035951747670475 0.6407421349445884;
+            -0.4649907741370211 0.7382129928076485;
+            -0.15772708898883683 0.25040532268222715;
+            -0.38144358562952446 0.6055745630707645;
+            -0.1012787681395544 0.16078875117643368;
+            -0.2741403589052255 0.4352214984054432];
+
+julia> y = X * [5, 10];
+
+julia> md1 = lm(X, y; method=:qr, dropcollinear=false)
+LinearModel
+
+Coefficients:
+───────────────────────────────────────────────────────────────────
+    Coef.  Std. Error             t  Pr(>|t|)  Lower 95%  Upper 95%
+───────────────────────────────────────────────────────────────────
+x1    5.0  2.53054e-8  197586428.62    <1e-63        5.0        5.0
+x2   10.0  1.59395e-8  627370993.89    <1e-99       10.0       10.0
+───────────────────────────────────────────────────────────────────
+
+julia> md2 = lm(X, y; method=:cholesky, dropcollinear=false)
```

Comment:
do you understand why when `dropcollinear=false` is dropped `:cholesky` produces a different result? Does `:qr` produce the same or different result?

Reply:
```
julia> using LinearAlgebra, GLM

julia> X = [-0.4011512997627107 0.6368622664511552;
                   -0.0808472925693535 0.12835204623364604;
                   -0.16931095045225217 0.2687956795496601;
                   -0.4110745650568839 0.6526163576003452;
                   -0.4035951747670475 0.6407421349445884;
                   -0.4649907741370211 0.7382129928076485;
                   -0.15772708898883683 0.25040532268222715;
                   -0.38144358562952446 0.6055745630707645;
                   -0.1012787681395544 0.16078875117643368;
                   -0.2741403589052255 0.4352214984054432];

julia> y = X * [5, 10];

julia> md1 = lm(X, y; method=:qr, dropcollinear=false)
LinearModel

Coefficients:
───────────────────────────────────────────────────────────────────
    Coef.  Std. Error             t  Pr(>|t|)  Lower 95%  Upper 95%
───────────────────────────────────────────────────────────────────
x1    5.0  2.53054e-8  197586428.62    <1e-63        5.0        5.0
x2   10.0  1.59395e-8  627370993.89    <1e-99       10.0       10.0
───────────────────────────────────────────────────────────────────


julia> md1 = lm(X, y; method=:qr, dropcollinear=true)
LinearModel

Coefficients:
────────────────────────────────────────────────────────────────────
    Coef.  Std. Error              t  Pr(>|t|)  Lower 95%  Upper 95%
────────────────────────────────────────────────────────────────────
x1    5.0  7.70812e-9   648666882.71    <1e-99        5.0        5.0
x2   10.0  4.85524e-9  2059629237.94    <1e-99       10.0       10.0
────────────────────────────────────────────────────────────────────


julia> md1 = lm(X, y; method=:cholesky, dropcollinear=false)
LinearModel

Coefficients:
────────────────────────────────────────────────────────────────
       Coef.  Std. Error       t  Pr(>|t|)  Lower 95%  Upper 95%
────────────────────────────────────────────────────────────────
x1   5.28865   0.103248    51.22    <1e-10    5.05056    5.52674
x2  10.1818    0.0650348  156.56    <1e-14   10.0318    10.3318
────────────────────────────────────────────────────────────────


julia> md1 = lm(X, y; method=:cholesky, dropcollinear=true)
LinearModel

Coefficients:
───────────────────────────────────────────────────────────────
       Coef.  Std. Error      t  Pr(>|t|)  Lower 95%  Upper 95%
───────────────────────────────────────────────────────────────
x1   5.64706    0.23454   24.08    <1e-08    5.10621    6.18791
x2  10.4076     0.147734  70.45    <1e-11   10.0669    10.7482
───────────────────────────────────────────────────────────────
```

Reply:
This is what I expected. So we get a different result in all four cases. Do you understand why?

Or maybe one of the maintainers of GLM.jl knows? I would expect that if we do not drop anything then `dropcollinear` setting should not have any impact on results.

CC @nalimilan @andreasnoack

Reply:
Unlike Cholesky, estimates are not changed for QR due to pivoting. Although, there are (not very significant) mismatch in std error. 

 The difference in factorization ( and hence the difference in inverse of X'X) with Cholesky might be the reason the difference in estimates.
```
julia> GLM.inverse(md11.pp)
2×2 Matrix{Float64}:
 2.52043e14  1.58759e14
 1.58759e14  1.0e14

julia> GLM.inverse(md12.pp)
2×2 Matrix{Float64}:
 2.52043e14  1.58759e14
 1.58759e14  1.0e14

julia> GLM.inverse(md21.pp)
2×2 Matrix{Float64}:
 2.57977e14  1.62497e14
 1.62497e14  1.02355e14

julia> GLM.inverse(md22.pp)
2×2 Matrix{Float64}:
 2.64918e14  1.66868e14
 1.66868e14  1.05108e14

julia> GLM.inverse(md11.pp) ≈ GLM.inverse(md12.pp)
true

julia> GLM.inverse(md21.pp) ≈ GLM.inverse(md22.pp)
false
```

Reply:
More over, at least in this example, Cholesky perform better (more close to original, and less standard error) than pivoting Cholesky.

Reply:
Yes, but why do we have differences in standard errors under `:qr`? This is what I do not understand. (we should resolve this in this PR I think, or at least understadnd the reason)

Also why do we have a different factorization in `:cholesky` if we in the end keep the variables (this is a question unrelated to this PR if we do not know).

Reply:
Is the difference in both cases beause we do/do not to pivoting respectively?

Reply:
> Yes, but why do we have differences in standard errors under `:qr`? This is what I do not understand. (we should resolve this in this PR I think, or at least understadnd the reason)

The reason is the differences in corresponding estimated dispersion/deviance and these are calculated from residuals 
 ```
julia> GLM.dispersion(md11)
1.5939531933641094e-15

julia> GLM.dispersion(md12)
4.855242786854203e-16

julia> deviance(md11)
2.0325494261085132e-29

julia> deviance(md12)
1.8858706015439813e-30

julia> residuals(md11)
10-element Vector{Float64}:
 -1.7763568394002505e-15
 -3.3306690738754696e-16
 -1.5543122344752192e-15
 -1.7763568394002505e-15
 -1.7763568394002505e-15
 -1.7763568394002505e-15
 -1.1102230246251565e-15
 -1.7763568394002505e-15
 -8.881784197001252e-16
  0.0

julia> residuals(md12)
10-element Vector{Float64}:
  0.0
 -1.1102230246251565e-16
 -2.220446049250313e-16
 -8.881784197001252e-16
  0.0
  0.0
  2.220446049250313e-16
  8.881784197001252e-16
  0.0
  4.440892098500626e-16
```

Reply:
But why do we have different residuals if we have the same estimates of coefficients?

Reply:
Actually,
```
julia> coef(md11)
2-element Vector{Float64}:
 4.999999976882807
 9.999999985438782

julia> coef(md12)
2-element Vector{Float64}:
  5.0000000040879975
 10.000000002574977

julia> isapprox(coef(md11), coef(md12))
true

julia> isapprox(coef(md11), coef(md12); atol=1.0e-16)
false

julia> isapprox(coef(md11), coef(md12); atol=1.0e-8)
false

julia> isapprox(coef(md11), coef(md12); atol=1.0e-7)
true
```

Reply:
I am digging into it since I wanted to know the exact reason. I think it is OK that they differ given the reason.

Reply:
Thanks @andreasnoack. we planned to have two examples where 

1. Cholesky detects multicollinearity eagerly like Wooldridge's example
2. QR estimates better in terms of precision 

There are many examples and issues related to Point 1.  After a lot of trial and error, finally able to create a dataset that reflects Point 2 (https://github.com/JuliaStats/GLM.jl/pull/507#issuecomment-1488860570). 

As you mentioned, I also thought to add some noise to `y` then we will have a decent standard error of estimates but estimates will differ from [ 5, 10].  Now I am thinking that we can compare the models based on likelihood values.

In that cases, the updated example will be 

```
julia> y = X*[5, 10] + 1.0e-08rand(10);

julia> mdqr = lm(X, y; method=:qr, dropcollinear=false)
LinearModel

Coefficients:
───────────────────────────────────────────────────────────────
      Coef.  Std. Error       t  Pr(>|t|)  Lower 95%  Upper 95%
───────────────────────────────────────────────────────────────
x1  4.98461   0.0571763   87.18    <1e-12    4.85276    5.11646
x2  9.99031   0.0360146  277.40    <1e-16    9.90726   10.0734
───────────────────────────────────────────────────────────────

julia> loglikelihood(mdqr)
181.34558612704268

julia> mdclol = lm(X, y; method=:cholesky, dropcollinear=false)
LinearModel

Coefficients:
───────────────────────────────────────────────────────────────
      Coef.  Std. Error       t  Pr(>|t|)  Lower 95%  Upper 95%
───────────────────────────────────────────────────────────────
x1  4.71135   0.113578    41.48    <1e-09    4.44944    4.97326
x2  9.81818   0.0715412  137.24    <1e-14    9.65321    9.98316
───────────────────────────────────────────────────────────────

julia> loglikelihood(mdchol)
174.59847167533468
```
@bkamins - looking for your opinion. 

Also, If we want to add Wooldridge's example, is there any better way to access the dataset other than putting the `csv` file in the data folder? - Please suggest.